src/visualization/utilities_p4est_t8code.jl

```diff
@@ -5,6 +5,303 @@
 @muladd begin
 #! format: noindent
 
+#############################################################################
+# 2D
+
+# The P4estMesh and T8codeMesh can make use of the efficient search functionality
+# based on the tree structure to speed up the process of finding the
+# elements in which the curve points are located.
+# We hard-code `Float64` here since some parts will be accessed from a Julia
+# function called as callback from C. Thus, we cannot use multiple dispatch
+# easily and use concrete types instead. See the comment on
+# `SearchPointsInP4estT8codeMesh2DHelper` below.
+# TODO: Implement the efficient search functionality also for the 2D T8codeMesh.
+function unstructured_2d_to_1d_curve(u, mesh::P4estMesh{2, 2, Float64},
+                                     equations, dg::DGSEM, cache,
+                                     curve, slice,
+                                     point, nvisnodes,
+                                     solution_variables)
+    # TODO: Currently, the efficient search functionality is only implemented
+    #       for linear meshes. If the mesh is not linear, we fall back to the naive
+    #       but general approach.
+    if length(mesh.nodes) != 2
+        return unstructured_2d_to_1d_curve_general(u, mesh, equations,
+                                                   dg, cache,
```

Comment:
The `dg` <-> `solver` issue also needs to be fixed in the 3D version (this line is untested), but I cannot comment there.

Reply:
Thanks! Fixed in https://github.com/trixi-framework/Trixi.jl/pull/2340/commits/91b414110f8c10240ec7c071e9f55197681abf92